test/spectests/spectests.cpp

```diff
@@ -58,6 +58,22 @@ fizzy::bytes load_wasm_file(const fs::path& json_file_path, std::string_view fil
         std::istreambuf_iterator<char>{wasm_file}, std::istreambuf_iterator<char>{});
 }
 
+bool float_values_equal(fizzy::Value value1, fizzy::Value value2) noexcept
+{
+    if (std::isnan(value1.as<float>()))
+        return value1.as<uint32_t>() == value2.as<uint32_t>();  // compare binary representations
```

Comment:
This is quite close to undefined behavior: only single union member may be active at any point in time and you are allowed to only access the active one. Here you are accessing two members is a row - one of the accesses must be wrong.

But also to simplify: you can always compare binary representation of floats. I would just leave `return value1.as<uint32_t>() == value2.as<uint32_t>();`.

Also: add the following check in `value.hpp`: `static_assert(offsetof(Value, f32) == offsetof(Value, i32));`.

Reply:
(now I see why you needed `to_uint64`/`from_uint64`)

Well ok, then the current way here it will be UB only for NaN case, but if I do always `as<uint32_t>()` then always...

Reply:
That's actually not true, because you always set `i64` to active in `read_value()`. In theory, you are never allowed to access `f32` nor `f64`. But I doubt this will cause issues in practice. However, I had issues with performing type punning using unions in GCC before (when the misuse is obvious, in the same function).

But my motivation to always use "binary" representation for comparison is to make sure this is a reliable method and there are no regressions in running spectests.

Reply:
This was replaced to comparison using `FP` utility from #454